spec/unit/shoulda/matchers/active_model/validate_numericality_of_matcher_spec.rb

```diff
@@ -69,26 +69,34 @@ def all_qualifiers # rubocop:disable Metrics/MethodLength
           validation_name: :on,
           validation_value: :customizable,
         },
+        {
+          category: :range,
+          name: :is_in,
+          argument: 1..10,
+          validation_name: :in,
+          validation_value: 1..10,
+          rails_version: 7.0,
+        },
       ]
     end
 
     def qualifiers_under(category)
-      all_qualifiers.select do |qualifier|
+      all_available_qualifiers.select do |qualifier|
         qualifier[:category] == category
       end
     end
 
     def mutually_exclusive_qualifiers
-      qualifiers_under(:cardinality) + qualifiers_under(:comparison)
+      qualifiers_under(:cardinality) + qualifiers_under(:comparison) + qualifiers_under(:range)
     end
 
     def non_mutually_exclusive_qualifiers
-      all_qualifiers - mutually_exclusive_qualifiers
+      all_available_qualifiers - mutually_exclusive_qualifiers
     end
 
-    def validations_by_qualifier
-      all_qualifiers.each_with_object({}) do |qualifier, hash|
-        hash[qualifier[:name]] = qualifier[:validation_name]
+    def all_available_qualifiers
+      all_qualifiers.filter do |qualifier|
+        rails_version >= qualifier.fetch(:rails_version, rails_oldest_version_supported)
       end
     end
```

Comment:
This method was not being used at all, I've deleted it, but happy to revert if it's needed in some way or not in the scope of this PR to make this change.